scripts/release-test-script-platform.sh

```diff
@@ -0,0 +1,242 @@
+sleep 10
+
+# Stop the command
+kill -9 "$command_pid" || error_exit "Failed to stop the command."
+
+sleep 10
+
+# Remove ut-core
+rm -rf ut/ut-core
+echo -e "${YELLOW}Removing ut-core so that, next build will download the correct branch${NC}"
+if [ -d ut/ut-core ]; then
+    echo -e "${YELLOW}ut-core was not removed, ensuring its removed${NC}"
+    rm -rf ut/ut-core
+fi
+
+sleep 5
+# Edit ut-control branch if provided
+if [ -n "$ut_core_branch_name" ]; then
+    sed -i "62s/.*/    git checkout feature\/$ut_core_branch_name/" ut/build.sh
+fi
+sleep 5
+
+echo -e "${YELLOW}You may also tail the output in another shell using ${NC}"
+echo -e "${MAGENTA}tail -f 100 $PWD/build_log.txt ${NC}"
+build_output=$(./build_ut.sh TARGET="$compiler_type" >> build_log.txt 2>&1)
+
+# # Check the build log for "Error"
+# grep_output=$(grep -nr "Error " build_log.txt )
+
+# # Print the results of the grep command
+# if [ -n "$grep_output" ]; then
+#     echo -e "${RED}Errors found in build_log.txt:${NC}"
+#     echo "$grep_output"
+#     exit 1
+# else
+#     echo -e "${GREEN}No unexpected errors found in build_log.txt${NC}"
+#     echo -e "${GREEN}build_ut command executed successfully.${NC}"
+# fi
+
+popd > /dev/null
+
+pushd ${PLAT_DIR} > /dev/null
+echo -e "current path during test is : $PWD"
+echo -e "${RED}==========================================================${NC}"
+CURL_STATIC_LIB="ut/ut-core/framework/ut-control/framework/curl/curl-8.8.0/build/lib/libcurl.a"
+OPENSSL_STATIC_LIB="ut/ut-core/framework/ut-control/framework/openssl/openssl-OpenSSL_1_1_1w/build/lib/libssl.a"
+CMAKE_HOST_BIN="ut/ut-core/framework/ut-control/host-tools/CMake-3.30.0/build/bin/cmake"
+
+echo -e "${RED}RESULTS for ${environment} ${NC}"
+
+grep_output=$(grep -nr "${ut_core_branch_name}" ut/build.sh)
+if [ -n "$grep_output" ]; then
+     echo -e "${GREEN}on the branch $ut_core_branch_name. PASS${NC}"
+else
+    error_exit "but, branch is not switched. FAIL"
+fi
+
+# Test for CURL
+if [ -f "$CURL_STATIC_LIB" ]; then
+    echo -e "${GREEN}$CURL_STATIC_LIB exists. PASS${NC}"
+else
+    echo -e "${RED}CURL static lib does not exist. FAIL ${NC}"
+fi
+
+# Test for Openssl
+if [ "$environment" == "ubuntu" ]; then
+    if [ ! -f "$OPENSSL_STATIC_LIB" ]; then
+        echo -e "${GREEN}Openssl static lib does not exist.PASS ${NC}"
+    else
+        echo -e "${RED}Openssl static lib does exist.FAIL ${NC}"
+    fi
+    elif [ "$environment" == "VM-SYNC" ]; then
+    if [ -f "$OPENSSL_STATIC_LIB" ]; then
+        echo -e "${GREEN}$OPENSSL_STATIC_LIB does exist.PASS ${NC}"
+    else
+        echo -e "${RED}Openssl static lib does not exist.FAIL ${NC}"
+    fi
+    elif [ "$environment" == "dunfell" ]; then
+    if [ -f "$OPENSSL_STATIC_LIB" ]; then
+        echo -e "${GREEN}$OPENSSL_STATIC_LIB does exist.PASS ${NC}"
+    else
+        echo -e "${RED}Openssl static lib does not exist.FAIL ${NC}"
+    fi
+fi
+
+# Test for CMAKE HOST Binary
+if [ "$environment" == "ubuntu" ]; then
```

Comment:
I don't think you should be switching here, it's all the same for everyone..

All these binaries should exist, and they should come from either the system or be built, and you should confirm where in whatever case, the requirement is that they  exist to meet your requirements.